test/NITTA/LuaFrontend/Tests.hs

```diff
@@ -63,19 +149,102 @@ test_lua_two_name_for_same_constant =
                 end
                 sum(0)
             |]
-     in -- TODO: it is a correct code and should be translated to dataflow graph
-        -- correctly, but in the current LuaFrontend stage, it cannot be done
-        -- without a huge redesign of LuaFrontend
-        [ expectFail $
-            testCase "contant with same name in lua source code" $ do
-                catch
-                    ( do
-                        let !_ = functions $ frDataFlow $ lua2functions src :: [F String Int]
-                        return ()
-                    )
-                    (\(_ :: ErrorCall) -> assertFailure "fail")
-                assertBool "check temporal restriction" True
-        ]
+        dfg =
+            [ constant 1 ["t^0#0"] :: F T.Text Int
+            , constant 1 ["r^0#0"]
```

Comment:
how can we merge this variables?
As I see, we don't have a possibility to send 3 args to Add func

Reply:
It is not a correct example from my side. In Lua, we can write something like `a + t + r + 1`. It will translate into 3 `add` functions, which should be refactored in 1 `Acc` function or optimized by constant folding.

See `NITTA.Intermediate.Functions.Accum`

Or I miss you question?